### PR TITLE
 Add Upgrade Guide to Apache Pulsar

### LINKED ISSUE #2483: Apache Pulsar Version Upgrade query
what's the most reliable method for Pulsar Upgrade from one dot release to another. 
For instance apache-pulsar-2.0.0-rc1-incubating to pulsar-2.1.0-incubating?
Any documentation or help is highly appreciated.

#### System configuration
**Pulsar version**: x.y

### LINKED ISSUE #4711: [Doc]Documentation on upgrading Pulsar is missing
Upgrading guide is missing, these are two items related to cluster upgrading:
-  https://github.com/apache/pulsar/commit/197e22c50ed4c09a550705259ddbb3e8eed10894
- https://github.com/apache/pulsar/blob/master/faq.md#is-there-any-sort-of-order-of-operations-or-best-practices-on-the-upgrade-procedure-for-a-geo-replicated-pulsar-cluster

Any plans on merging these two and update in the doc?

cc @sijie @Anonymitaet

### PR DESCRIPTION
**Pull request by sijie**
### Motivation

Add an upgrade guide for Pulsar.

Fixes #4711 
Fixes #2483

### CODE REVIEW

**Comment by sijie** [comment]
@Anonymitaet please help review this pull request.

File: site2/docs/administration-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+### Canary Test
+
+Although they are `stateless` and easier to upgrade, it is still wise to verify a new version in one or small set of nodes before upgrading
+all nodes in your live cluster.
+
+You can follow below steps on how to canary a upgraded version:
+
+1. Stop a Broker (or Proxy).
+2. Upgrade the binary and configuration.
+3. Start a Broke (or Proxy)r
```
Comment (yjshen) [inline site2/docs/administration-upgrade.md]:
```suggestion
3. Start a Broker (or Proxy).
```

File: site2/docs/administration-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+4. Once the Bookie is running at `ReadOnly` mode successfully for a while, stop the bookie and restart it in `Write/Read` mode.
+   ```shell
+   bin/pulsar bookie
+   ```
+5. After step 4, the Bookie will serve both write and read traffic.
+
+#### Canary Rollback
+
+If problems occur during canarying an upgraded version, you can simply take down the problematic Bookie node. The remain bookies in the old cluster
+will repair this problematic bookie node by autorecovery. Nothing needs to be worried about.
```
Comment (yjshen) [inline site2/docs/administration-upgrade.md]: Does it assume autorecovery is enabled? Is it a standard configuration for bookies? What if it's disabled? 

Does it mean all the ledgers in this bookie are replicated to the left nodes? What should I do if I don't want to shrink the size of my bookie cluster? Should I manually delete all the data in this node and added it back to bookie cluster then?

**Comment by sijie** [comment]
@yjshen I will address your comments in a separated pull request. since it was merged.

**Pull request merged**